### PR TITLE
Add `ServerBuilder.localPort` shortcuts

### LINKED ISSUE #3599: Add `ServerBuilder.loopbackPort(int)`
A user often binds on a loopback port to serve internal use only requests. It is currently done with `port()` method:

```java
Server
  .builder()
  .port(new InetSocketAddress("127.0.0.1", 8080))
  ...
```

It could be useful to have a shortcut method `loopbackPort(int)`:

```java
Server
  .builder()
  .loopbackPort(8080)
  .loopbackPort(9090, SessionProtocol.HTTP, SessionProtocol.HTTPS)
  .loopbackPort(10100, Set.of(SessionProtocol.HTTP, SessionProtocol.HTTPS))
  ...
```

### PR DESCRIPTION
**Pull request by kezhenxu94**
Closes #3599

### CODE REVIEW

**Review by ikhoon — APPROVED** [review]
Blazing speed! Thanks, @kezhenxu94

File: core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java [hunk 1]
```diff
@@ -364,6 +364,45 @@ public ServerBuilder port(ServerPort port) {
         return this;
     }
 
+    /**
+     * Adds the specified {@link ServerPort} to loopback {@code localAddress}.
+     *
+     * @see <a href="#no_port_specified">What happens if no HTTP(S) port is specified?</a>
+     */
+    public ServerBuilder loopbackPort(int port) {
+        return port(new InetSocketAddress("127.0.0.1", port));
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java:373]: Could use `NetUtil.LOCALHOST` instead?

Comment (kezhenxu94) [reply core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java]: > Could use `NetUtil.LOCALHOST` instead?

Exactly 🙇

File: core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java [hunk 2]
```diff
@@ -364,6 +365,45 @@ public ServerBuilder port(ServerPort port) {
...
     }
 
+    /**
+     * Adds the specified {@link ServerPort} to loopback {@code localAddress}.
+     *
+     * @see <a href="#no_port_specified">What happens if no HTTP(S) port is specified?</a>
+     */
+    public ServerBuilder loopbackPort(int port) {
+        return port(new InetSocketAddress(NetUtil.LOCALHOST, port));
+    }
```
Comment (ikhoon) [inline core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java:375]: How about removing this method in favor of `loopbackPort(int port, SessionProtocol... protocols)`?

File: core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java [hunk 3]
```diff
@@ -364,6 +365,45 @@ public ServerBuilder port(ServerPort port) {
...
+     *
+     * <pre>{@code
+     * ServerBuilder sb = Server.builder();
+     * sb.loopbackPort(8080,
+     *         Arrays.asList(SessionProtocol.HTTP,
+     *                       SessionProtocol.HTTPS));
+     * }</pre>
+     */
+    public ServerBuilder loopbackPort(int port, Iterable<SessionProtocol> protocols) {
+        return port(new InetSocketAddress("127.0.0.1", port), protocols);
```
Comment (ikhoon) [inline core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java:404]:
```suggestion
        return port(new InetSocketAddress(NetUtil.LOCALHOST, port), protocols);
```

File: core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java [hunk 4]
```diff
@@ -364,6 +365,45 @@ public ServerBuilder port(ServerPort port) {
...
+     *
+     * <pre>{@code
+     * ServerBuilder sb = Server.builder();
+     * sb.loopbackPort(8080,
+     *         Arrays.asList(SessionProtocol.HTTP,
+     *                       SessionProtocol.HTTPS));
+     * }</pre>
+     */
+    public ServerBuilder loopbackPort(int port, SessionProtocol... protocols) {
+        return port(new InetSocketAddress("127.0.0.1", port), protocols);
```
Comment (ikhoon) [inline core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java:389]:
```suggestion
        requireNonNull(protocols, "protocols");
        return loopbackPort(port, ImmutableList.copyOf(protocols));
```

**Review by ikhoon — COMMENTED** [review]
By the way how about renaming `loopbackPort` to `localPort` which is shorter with the same meaning. https://github.com/line/armeria/issues/3599#issuecomment-855189887

**Review by minwoox — APPROVED** [review]
Thanks!

File: core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java [hunk 5]
```diff
@@ -364,6 +365,37 @@ public ServerBuilder port(ServerPort port) {
...
     }
 
+    /**
+     * Adds a new {@link ServerPort} that listens to the loopback {@code localAddress} using the specified
+     * {@link SessionProtocol}s. Specify multiple protocols to serve more than one protocol on the same port:
+     *
+     * <pre>{@code
+     * ServerBuilder sb = Server.builder();
+     * sb.localPort(8080,
+     *         Arrays.asList(SessionProtocol.HTTP,
```
Comment (ikhoon) [inline core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java:375]: Use varargs? 😆

Comment (kezhenxu94) [reply core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java]: > Use varargs? 😆

😅 sorry for so hasty, now done

**Review by anuraaga — APPROVED** [review]

File: core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java [hunk 6]
```diff
@@ -364,6 +365,33 @@ public ServerBuilder port(ServerPort port) {
...
+        return localPort(port, ImmutableList.copyOf(protocols));
+    }
+
+    /**
+     * Adds a new {@link ServerPort} that listens to the loopback {@code localAddress} using the specified
+     * {@link SessionProtocol}s. Specify multiple protocols to serve more than one protocol on the same port:
+     *
+     * <pre>{@code
+     * ServerBuilder sb = Server.builder();
+     * sb.localPort(8080, SessionProtocol.HTTP, SessionProtocol.HTTPS);
```
Comment (ikhoon) [inline core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java:388]: It should be a list. 😉
```suggestion
     * sb.localPort(8080, List.of(SessionProtocol.HTTP, SessionProtocol.HTTPS));
```

**Review by trustin — APPROVED** [review]

**Pull request merged**